src/RageTimer.cpp

```diff
@@ -29,13 +29,19 @@
 
 #include <cmath>
 #include <cstdint>
+#include <atomic>
 
 const std::uint64_t ONE_SECOND_IN_MICROSECONDS_ULL = 1000000ULL;
 const std::int64_t ONE_SECOND_IN_MICROSECONDS_LL = 1000000LL;
 const double ONE_SECOND_IN_MICROSECONDS_DBL = 1000000.0;
 
 const RageTimer RageZeroTimer(0,0);
-static std::uint64_t g_iStartTime = ArchHooks::GetMicrosecondsSinceStart( true );
+
+// Global atomic start time for initialization
+static std::atomic<std::uint64_t> g_iGlobalStartTime = ArchHooks::GetMicrosecondsSinceStart(true);
+
+// Thread-local start time
+thread_local std::uint64_t g_iThreadStartTime = g_iGlobalStartTime.load();
```

Comment:
If this is always g_iGlobalStartTime, and we never actually modify g_iGlobalStartTime or g_iThreadStartTime, do we even need the thread-local variable?